components/fastly_backends/main.tf

```diff
@@ -3,8 +3,8 @@ variable "name" {
   type        = string
 }
 
-variable "backends" {
-  description = "The domains/backends to route traffic to in this property."
+variable "applications" {
+  description = "The application modules to route traffic to."
```

Comment:
I renamed the `backends` argument provided to `applications` for two reasons:

1. It's odd to provide a "backends" object that itself is composed of name, domain, and backend(!) properties.
2. I wanted to follow the same convention (`applications = […]` and `application = …`) for both these Fastly modules! It would feel somewhat awkward to pass a `backend` argument to the frontend module! 😂